firmware/library/L1_Drivers/can.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "L0_LowLevel/LPC40xx.h"
+#include "L2_Utilities/macros.hpp"
+
+SJ2_PACKED(struct) CanMessage_t
```

Comment:
The reason I put `CanMessage_t` in a global scope is so that the user can define a message where ever they want, and when the user wants to send/receive they pass their struct instance to `Send(const CanMessage_t &message)` and `Receive(canMessage_t * message)`.

Are you suggesting that driver should just have one local instance of CanMessage_t for it's own processing, and then have the user send id, data, etc through `Send(int id, int data, etc)`?

Reply:
> The reason I put CanMessage_t in a global scope
Allowing the user to be able to access the message doesn't require it to be global scope. You could put the `CanMessage_t` in the class scope, just make it public. So the user can do the following:
```
CanInterface::Message_t user_defined_can_message;
```

Reply:
Ok, nice. I didn't know you could do that with structs. I'll make that change then.

Reply:
moved into CanInterface